pkg/cmd/pulumi/registry/package_resolution.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"context"
+	"errors"
+	"path/filepath"
+
+	"github.com/blang/semver"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/registry"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+type PackageFallbackType int
+
+const (
+	NoFallback                PackageFallbackType = iota
+	PreGitHubRegistryFallback                     // Use GitHub releases for pre-IDP Registry packages
+	LocalProjectFallback                          // Use local Pulumi.yaml package definition
```

Comment:
Shouldn't this be an override, not a fallback? I assume this takes precedence over how other methods.